synapse/handlers/federation.py

```diff
@@ -786,13 +801,18 @@ def backfill(self, dest, room_id, limit, extremities):
             })
 
         for e_id in events_to_state:
+            # For paranoia we ensure that these events are marked as
```

Comment:
this seems to be in conflict with the comment at line 775 that says "... state get persisted as outliers". What am I missing?

Reply:
This needs more comments, obviously. The keys of `events_to_state` are the new backwards extremities, as those are the events that we fetched state for

Reply:
it would seem a whole lot less opaque to iterate over `edges` rather than the keys of `events_to_state`, but ok.